tools/jira_notificator.py

```diff
@@ -0,0 +1,164 @@
+    return f"{create_notification_prefix(NotificationType.QA_CONTACT)} {qa_contact_link}Please verify the Jira as soon as possible."
+
+def notify_qa_contact(jira: JIRA, issue: Issue) -> None:
+    jira.add_comment(issue, create_qa_notification_text(issue))
+
+def create_team_lead_notification_text(issue: Issue) -> str:
+    qa_contact = get_qa_contact(issue)
+    qa_contact_link = f"[~{qa_contact.name}] " if qa_contact else ""
+    return f"{create_notification_prefix(NotificationType.TEAM_LEAD)} {qa_contact_link}Please verify the Jira as soon as possible or arrange a reassignment with your team lead."
+
+def notify_team_lead(jira: JIRA, issue: Issue) -> None:
+    logger.info("Notification to the team lead is not yet implemented. Notifying the QA contact again.")
+    jira.add_comment(issue, create_team_lead_notification_text(issue))
+
+def create_manager_notification_text(jira: JIRA, issue: Issue) -> str:
+    qa_contact = get_qa_contact(issue)
+    manager_link = ""
+    if qa_contact:
+        manager = get_manager(jira, qa_contact)
+        manager_link = f"[~{manager.name}] " if manager else ""
+    else:
+        logger.warning("Notification to QA Contact Manager could not be sent because of missing QA contact.")
+    return f"{create_notification_prefix(NotificationType.MANAGER)} {manager_link}Please prioritize the verification or consider reassigning it to another available QA Contact."
+
+def notify_manager(jira: JIRA, issue: Issue) -> None:
+    jira.add_comment(issue, create_manager_notification_text(jira, issue))
+   
+def get_latest_on_qa_transition_datetime(issue: Issue) -> Optional[datetime]:
+    latest_on_qa: datetime = None
+
+    for history in issue.changelog.histories:
+        for item in history.items:
+            if item.field == "status" and item.toString == "ON_QA":
+                transition_time = parser.parse(history.created)
+                if not latest_on_qa or transition_time > latest_on_qa:
+                    latest_on_qa = transition_time
+
+    if not latest_on_qa:
+        logger.error(f"Issue {issue.key} does not have ON_QA transition date")
+
+    return latest_on_qa
+
+def get_latest_ert_notification_type_after_on_qa_transition(issue: Issue, on_qa_transition: datetime) -> Optional[NotificationType]:
+    latest_notification_type = None
+    latest_datetime = None
+
+    for comment in issue.fields.comment.comments:
+        if not comment.body.startswith(ERT_NOTIFICATION_PREFIX):
+            continue
+
+        created_datetime = parser.parse(comment.created)
+        if created_datetime < on_qa_transition:
+            continue
+
+        if not latest_datetime or created_datetime > latest_datetime:
+            latest_datetime = created_datetime
+            latest_notification_type = get_notification_type(comment.body)
+
+    return latest_notification_type
+
+
+def check_issue_and_notify_responsible_people(jira: JIRA, issue: Issue) -> None:
```

Comment:
I suggest handling errors in this func. to avoid the case that any unexpected error occurs, the process will be break